Composer/packages/client/src/store/action/lu.ts

```diff
@@ -1,49 +1,37 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 import clonedeep from 'lodash/cloneDeep';
-import debounce from 'lodash/debounce';
 
 import * as luUtil from '../../utils/luUtil';
 import { undoable } from '../middlewares/undo';
-import { ActionCreator, State } from '../types';
+import { ActionCreator, State, Store } from '../types';
 import luFileStatusStorage from '../../utils/luFileStatusStorage';
 
 import httpClient from './../../utils/httpUtil';
 import { ActionTypes } from './../../constants/index';
-import { setError } from './error';
-//remove editor's debounce and add it to action
-export const debouncedUpdateLu = debounce(async (store, id, projectId, content, lastModified) => {
-  try {
-    const response = await httpClient.put(`/projects/${projectId}/luFiles/${id}`, {
-      id,
-      projectId,
-      content,
-      lastModified,
-    });
-    store.dispatch({
-      type: ActionTypes.UPDATE_TIMESTAMP,
-      payload: {
-        id: id,
-        type: 'lu',
-        lastModified: response.data.lastModified,
-      },
-    });
-  } catch (err) {
-    setError(store, {
-      status: err.response.status,
-      message: err.response && err.response.data.message ? err.response.data.message : err,
-      summary: 'UPDATE LU ERROR',
-    });
-  }
-}, 500);
 
-export const updateLuFile: ActionCreator = async (store, { id, projectId, content }) => {
-  const file = store.getState().lgFiles.find(l => l.id === id);
-  if (file) {
-    luFileStatusStorage.updateFile(store.getState().botName, id);
-    store.dispatch({ type: ActionTypes.UPDATE_LU_SUCCESS, payload: { id, content } });
-    debouncedUpdateLu(store, id, projectId, content, file.lastModified);
-  }
+export const updateLuFile: ActionCreator = async (store, { id, content }) => {
+  luFileStatusStorage.updateFile(store.getState().botName, id);
+  store.dispatch({
+    type: ActionTypes.UPDATE_LU,
+    payload: { id, content },
+  });
+};
```

Comment:
Correct pattern:
```ts
// Action
const action = updateLuFile('a.lu'); // got { type: 'UPDATE_LU', payload: { id: 'a.lu' } }

// Reducer
const reducer = (store, action) => {
   if (action.type === 'UPDATE_LU') {
      const newLuFile = update(action.id);
      const luFiles = replaceOldLuFileWithNewLuFile(store.luFiles, 'a.lu', newLuFile);
      return { ...store, luFiles };
   }
}
```
Reason: Decouple your core domain events from bussiness logic.
`Action` is your core domain events, it connects UI and store; meanwhile `Reducer` is the place you handle the bussiness logic, it varies.

Reply:
Though we prefer `useReducer` over `redux`, the redux pattern should still be followed.